### PR TITLE
[Packager] Changes to conversion of modules from Windows directory to standard convention

### PR DESCRIPTION
**Pull request by joewood**
Related to issue #468 - Enabled Packager to run on Windows
This change relates to the haste. Module IDs are converted from the Windows Path convention to standard convention for module-ids and dependencies. The code detects the running OS using the standard node.js `os` module. If not running on Windows the `convertPath` function is set to a no-op.

### CODE REVIEW

**Comment by joewood** [comment]
Looks like the Travis CI server doesn't support `os.type()`.

File: packager/react-packager/src/DependencyResolver/haste/index.js [hunk 1]
```diff
@@ -157,10 +168,10 @@ HasteDependencyResolver.prototype.wrapModule = function(module, code) {
 
   return DEFINE_MODULE_CODE.replace(DEFINE_MODULE_REPLACE_RE, function(key) {
     return {
-      '_moduleName_': module.id,
+      '_moduleName_': convertPath(module.id),
       '_code_': code.replace(replacePatterns.IMPORT_RE, relativizeCode)
                     .replace(replacePatterns.REQUIRE_RE, relativizeCode),
-      '_deps_': JSON.stringify(resolvedDepsArr),
+      '_deps_': JSON.stringify(resolvedDepsArr.map(convertPath))
```
Comment (amasad) [inline packager/react-packager/src/DependencyResolver/haste/index.js]: I'm not convinced that we need to change runtime code to support windows. Does node, browserify, or windows change the generated output for windows?

If we fixed things internally we can generate the same output for windows or unix, i.e. the bundle could continue to refer to modules using `/`

**Comment by amasad** [comment]
Can you explain why we need to change the generated output? It's not really going to run on the filesystem, it's only in the browser. Why do we need to change the module names.

**Comment by joewood** [comment]
The module names were inconsistent, the dependencies used "\" and the defines used "/". The output was changed to make it all "/" to be consistent (it has no impact on Mac, as it never contains a backslash). This was the main problem with the packager.
A better fix may be to convert to forward slash at the source. I tried that, but had no luck.

**Comment by joewood** [comment]
I can try this PR by changing the paths at source, so the output isn't modified. If you prefer - I'll give it another go this weekend.

**Comment by amasad** [comment]
@joewood that'd be great. Having the same output regardless of the platform is great. Reproducible builds is a good virtue of build systems.

**Comment by marynate** [comment]
When I run packager on windows after apply this PR and the blacklist.js PR, I got this error in Xcode output (running simulator on Mac, requesting bundle on windows):

>  2015-04-24 18:09:00.367 [info][tid:com.facebook.React.JavaScript] "Failed to print error: ", "Requiring module "Promise" which threw an exception"
2015-04-24 18:09:00.372 [info][tid:com.facebook.React.JavaScript] "Failed to print error: ", "Requiring module "ExceptionsManager" which threw an exception"

**Comment by brentvatne** [comment]
@joewood - what's the status of this? is this still relevant?

**Comment by joewood** [comment]
Yes, this is still needed. I made a change to this PR to change the dir convention coming into the packager, rather than changing the packager output (as per @amasad suggestion).  I haven't looked at the test failures from the PR and need to rebase.

**Comment by amasad** [comment]
I can rebase and do some testing. Can you add a test plan? What version of windows does this support? and how do you test it? (can you run xcode etc on windows?)

**Comment by brentvatne** [comment]
ping @joewood ^

**Comment by satya164** [comment]
@joewood Any updates on this?

**Comment by brentvatne** [comment]
This PR seems dead, closing, will reopen if anyone shows signs of life!

**Pull request closed without merging**